src/node.cc

```diff
@@ -93,10 +94,26 @@ std::ostream &operator<<(std::ostream &stream, const AbstractNode &node)
 // Do we have an explicit root node (! modifier)?
 AbstractNode *find_root_tag(AbstractNode *n)
 {
-  for(auto v : n->children) {
-    if (v->modinst->tag_root) return v;
-    if (auto vroot = find_root_tag(v)) return vroot;
-  }
-  return NULL;
-}
+	std::vector<AbstractNode*> rootTags;
+
+	std::function <void (AbstractNode *n)> find_root_tags = [&] (AbstractNode *n)
```

Comment:
You should be able to use `auto` here instead of declaring it using std::function